core/services/workflows/engine.go

```diff
@@ -58,26 +46,34 @@ func (e *Engine) init(ctx context.Context) {
 	retrySec := 5
 	ticker := time.NewTicker(time.Duration(retrySec) * time.Second)
 	defer ticker.Stop()
+
+	// Note: in our hardcode workflow, there is only one trigger,
+	// and one consensus step.
+	trigger := e.workflow.Triggers[0]
+	consensus := e.workflow.Consensus[0]
+
 	var err error
 LOOP:
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
-			e.trigger, err = e.registry.GetTrigger(ctx, e.triggerType)
+			e.trigger, err = e.registry.GetTrigger(ctx, trigger.Type)
 			if err != nil {
 				e.logger.Errorf("failed to get trigger capability: %s, retrying in %d seconds", err, retrySec)
 				break
 			}
-			e.consensus, err = e.registry.GetConsensus(ctx, e.consensusType)
+
+			e.consensus, err = e.registry.GetConsensus(ctx, consensus.Type)
 			if err != nil {
```

Comment:
is the error handling correct? i see it break the LOOP on an error ,but i don't see a check for a non-nil error after LOOP. it looks like err is intermediately overwrittern

Reply:
Yes, I think the intent was to log the error but otherwise continue